### PR TITLE
add support for reference inputs

### PR DESCRIPTION
**Pull request by saratomaz**

### CODE REVIEW

**Review by mkoura — CHANGES_REQUESTED** [review]
This will need to be added to other places as well (`build_raw_tx_bare`, `build_raw_tx`, `calculate_tx_fee`). See e.g. `total_collateral_amount`, this will be the same situation.

File: cardano_clusterlib/clusterlib_klass.py [hunk 1]
```diff
@@ -2514,6 +2537,7 @@ def send_tx(
             tx_name=tx_name,
             txins=txins,
             txouts=txouts,
+            readonly_reference_txins=readonly_reference_txins,
```
Comment (mkoura) [inline cardano_clusterlib/clusterlib_klass.py:2540]: Also missing in `self.calculate_tx_fee` call above.

Comment (saratomaz) [reply cardano_clusterlib/clusterlib_klass.py:2541]: Updated. There we don't have the new fields related to collaterals, no need?

Comment (mkoura) [reply cardano_clusterlib/clusterlib_klass.py:2541]: Not sure if that new collateral stuff increase tx size / affect tx fee, we'll need to test that.

**Review by mkoura — CHANGES_REQUESTED** [review]
Also please add it to `structs.TxRawOutput`.

**Review by mkoura — APPROVED** [review]

**Pull request merged**